### DISCUSSION TITLE
Notification filters

### DISCUSSION THREAD

**Discussion by abdullah-kasim** [regolith-linux/regolith-desktop#835] [Ideas]
I'm wondering if we could add notification filters through a scriptable python file or similar.

Use case: 

Currently I'm receiving notifications from KDE Connect on my phone but I do not want to receive all of the notifications. It'd be nice if we'd be able to filter out the notifications that I don't need.

**Comment by marchank0** [comment]
Why do not to make it even more flexible?  
iOS has a focus functionality that allows to filter notifications based on active focus.  
So, it would be nice to implement focus activation based on some predefined criteria and allow to manage notifications inside the focus with some scriptable language like Python or Lua

**Comment by lawrencegripper** [comment]
I've built something like this in python using the existing libraries (it also calls a physical light on my desk).

Here's the code if you want to play around:

https://github.com/lawrencegripper/dotfiles/blob/main/private_dot_config/polybar/executable_notification_light.py
